### PR TITLE
Javascript: Add iFrame based Task Module sample

### LINKED ISSUE #2574: Add iFrame based Task Module samples for remaining languages
On the back of the .NET sample https://github.com/microsoft/BotBuilder-Samples/pull/2546 - add this sample to the other languages.

- [ ] Python (https://github.com/microsoft/BotBuilder-Samples/tree/trboehre/teamstasks)
- [ ] JS
- [x] Java

### PR DESCRIPTION
**Pull request by tracyboehrer**
Fixes #2574

### CODE REVIEW

**Comment by mrivera-ms** [comment]
This is related to the R10 task already implemented in .NET.

File: samples/javascript_nodejs/54.teams-task-module/bots/teamsTaskModuleBot.js [hunk 1]
```diff
@@ -2,63 +2,133 @@
...
-                type: 'continue',
-                value: {
-                    card: this.getTaskModuleAdaptiveCard(),
-                    height: 220,
-                    width: 400,
-                    title: 'Adaptive Card: Inputs'
-                }
+                type: 'message',
+                value: 'Thanks!'
             }
```
Comment (mdrichardson) [inline samples/javascript_nodejs/54.teams-task-module/bots/teamsTaskModuleBot.js:78]: Do we want to change this to:

```js
return TaskModuleResponseFactory.createResponse('Thanks!');
```

File: samples/javascript_nodejs/54.teams-task-module/bots/teamsTaskModuleBot.js [hunk 2]
```diff
@@ -2,63 +2,133 @@
 // Licensed under the MIT License.
 
 const { TeamsActivityHandler, MessageFactory, CardFactory } = require('botbuilder');
+const { TaskModuleUIConstants } = require('../models/TaskModuleUIConstants');
+const { TaskModuleIds } = require('../models/taskmoduleids');
+const { TaskModuleResponseFactory } = require("../models/taskmoduleresponsefactory");
```
Comment (mdrichardson) [inline samples/javascript_nodejs/54.teams-task-module/bots/teamsTaskModuleBot.js:7]:
```suggestion
const { TaskModuleResponseFactory } = require('../models/taskmoduleresponsefactory');
```

File: samples/javascript_nodejs/54.teams-task-module/bots/teamsTaskModuleBot.js [hunk 3]
```diff
@@ -2,63 +2,133 @@
...
-            [{ type: 'invoke', title: 'Task Module', value: { type: 'task/fetch', data: 'adaptivecard' } }]);
+        if (cardTaskFetchValue == TaskModuleIds.YouTube) {
+            // Display the YouTube.html page
+            taskInfo.url = taskInfo.fallbackUrl = this.baseUrl + "/" + TaskModuleIds.YouTube + ".html";
+            this.setTaskInfo(taskInfo, TaskModuleUIConstants.YouTube);
+        }
+        else if (cardTaskFetchValue == TaskModuleIds.CustomForm) {
+            // Display the CustomForm.html page, and post the form data back via
+            // handleTeamsTaskModuleSubmit.
+            taskInfo.url = taskInfo.fallbackUrl = this.baseUrl + "/" + TaskModuleIds.CustomForm + ".html";
```
Comment (mdrichardson) [inline samples/javascript_nodejs/54.teams-task-module/bots/teamsTaskModuleBot.js:52]:
```suggestion
            taskInfo.url = taskInfo.fallbackUrl = this.baseUrl + '/' + TaskModuleIds.YouTube + '.html';
            this.setTaskInfo(taskInfo, TaskModuleUIConstants.YouTube);
        }
        else if (cardTaskFetchValue == TaskModuleIds.CustomForm) {
            // Display the CustomForm.html page, and post the form data back via
            // handleTeamsTaskModuleSubmit.
            taskInfo.url = taskInfo.fallbackUrl = this.baseUrl + '/' + TaskModuleIds.CustomForm + '.html';
```

File: samples/javascript_nodejs/54.teams-task-module/index.js [hunk 4]
```diff
@@ -57,3 +57,8 @@ server.post('/api/messages', (req, res) => {
         await bot.run(context);
     });
 });
+
+// Serve static pages from the 'pages' folder.
+server.get("/*", restify.plugins.serveStatic({
```
Comment (mdrichardson) [inline samples/javascript_nodejs/54.teams-task-module/index.js:62]:
```suggestion
server.get('/*', restify.plugins.serveStatic({
```

File: samples/javascript_nodejs/54.teams-task-module/models/taskmoduleresponsefactory.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+class TaskModuleResponseFactory {
+    static createResponse(taskModuleInfoOrString) {
+        if (typeof taskModuleInfoOrString == "string") {
```
Comment (mdrichardson) [inline samples/javascript_nodejs/54.teams-task-module/models/taskmoduleresponsefactory.js:6]:
```suggestion
        if (typeof taskModuleInfoOrString == 'string') {
```

File: samples/javascript_nodejs/54.teams-task-module/models/taskmoduleuiconstants.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const { UISettings } = require("./uisettings");
+const { TaskModuleIds } = require("./taskmoduleids");
+
+const TaskModuleUIConstants = {
+    YouTube: new UISettings(1000, 700, "YouTube Video", TaskModuleIds.YouTube, "YouTube"),
+    CustomForm: new UISettings(510, 450, "Custom Form", TaskModuleIds.CustomForm, "Custom Form"),
+    AdaptiveCard: new UISettings(400, 200, "Adaptive Card: Inputs", TaskModuleIds.AdaptiveCard, "Adaptive Card")
```
Comment (mdrichardson) [inline samples/javascript_nodejs/54.teams-task-module/models/taskmoduleuiconstants.js:10]:
```suggestion
const { UISettings } = require('./uisettings');
const { TaskModuleIds } = require('./taskmoduleids');

const TaskModuleUIConstants = {
    YouTube: new UISettings(1000, 700, 'YouTube Video', TaskModuleIds.YouTube, 'YouTube'),
    CustomForm: new UISettings(510, 450, 'Custom Form', TaskModuleIds.CustomForm, 'Custom Form'),
    AdaptiveCard: new UISettings(400, 200, 'Adaptive Card: Inputs', TaskModuleIds.AdaptiveCard, 'Adaptive Card')
```

**Pull request merged**